packages/flutter/lib/src/services/deferred_component.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import 'system_channels.dart';
+
+/// Default Google Play Store implementation states that reflect the current
+/// installation status of a deferred component module.
+///
+/// Returned by [DeferredComponent.getDeferredComponentInstallState].
+///
+/// These states may differ in custom deferred component implementations.
```

Comment:
What makes a component "custom"? Where could the reader learn more about custom components?

Reply:
What would DeferredComponent.getDeferredComponentInstallState return for custom components that have different state?

Reply:
I'll be writing a guide on creating custom deferred components. Custom components are very open ended, and thus, this API has been left open ended. Since this state getting API is informational/quality-of-life, this open-ended-ness should not impact functionality, so I planned on leaving it as a channel to pass whatever information custom implementers wish. Also, custom implementations can be considered a very advanced feature, as it requires not only the custom impl and servers to host the files, but also a custom build setup to build the custom components.